tuning/scripts/provision_repo.sh

```diff
@@ -1,56 +1,64 @@
 #!/bin/bash
 
-# #################################################
-# Parameter parsing
-# #################################################
-
-WORKING_PATH='release'
 HELP=false
-ROCBLAS=false
+WORKING_PATH='release'
 PROVISION_BRANCH='develop'
-TENSILE_FORK='RocmSoftwarePlatform'
-ROCBLAS_FORK='RocmSoftwarePlatform'
+FORK='RocmSoftwarePlatform'
 
 GIT_HOST="${TENSILE_HOST}"
 PROVISION_PATH='Tensile'
 
-HELP_STR="usage: $0 [-b|--branch <branch>] [-f|--tensile-fork <username>] [-w|--working-path <path>] [-i <identifier>] [-t|--tag <githup tag>] [-h|--help]"
-
-OPTS=`getopt -o hf:rw:t:b:c:i: --long help,tensile-fork:,rocblas-fork:,working-path:,tag:,branch:,commit: -n 'parse-options' -- "$@"`
-
-if [ $? != 0 ] ; then echo "Failed parsing options." >&2 ; exit 1 ; fi
+HELP_STR="
+Usage: ${0} [options]
+
+Options:
+-h | --help                    Display this help message
+-r                             Provision rocBLAS instead of Tensile
+-w | --working-path PATH       Working path for tuning
+-f | --tensile-fork USERNAME   Fork to use
+-b | --branch BRANCH           Branch to use
+-c | --commit COMMIT_ID        Commit to use
+-t | --tag GITHUB_TAG          Tag to use
+-i | --id ID                   ID to append to directory name
+"
+
+if ! OPTS=$(getopt -o h,r,w:,f:,b:,c:,t:,i: \
+--long help,working-path:,fork:,branch:,commit:,tag:,id: -n 'parse-options' -- "$@")
+then
+  echo "Failed parsing options"
+  exit 1
+fi
 
-eval set -- "$OPTS"
+eval set -- "${OPTS}"
 
 while true; do
-  case "$1" in
+  case ${1} in
     -h | --help )         HELP=true; shift ;;
-    -f | --tensile-fork ) TENSILE_FORK="$2"; shift 2;;
-    --rocblas-fork )      ROCBLAS_FORK="$2"; shift 2;;
     -r )                  PROVISION_PATH='rocBLAS'; shift;;
-    -w | --working-path ) WORKING_PATH="$2"; shift 2;;
-    -t | --tag )          TAG="$2"; shift 3;;
-    -b | --branch  )      PROVISION_BRANCH="$2"; shift 2;;
-    -c | --commit )       COMMIT="$2"; shift 2;;
-    -i )                  ID="$2"; shift 2;;
+    -w | --working-path ) WORKING_PATH=${2}; shift 2;;
+    -f | --fork )         FORK=${2}; shift 2;;
+    -b | --branch  )      PROVISION_BRANCH=${2}; shift 2;;
+    -c | --commit )       COMMIT=${2}; shift 2;;
+    -t | --tag )          TAG=${2}; shift 2;;
+    -i | --id )           ID=${2}; shift 2;;
     -- ) shift; break ;;
     * ) break ;;
   esac
 done
 
-if $HELP; then
-  echo "${HELP_STR}" >&2
-  exit 2
+if ${HELP}; then
+  echo "${HELP_STR}"
+  exit 0
 fi
 
-mkdir -p ${WORKING_PATH}
+mkdir -p "${WORKING_PATH}"
 
-pushd ${WORKING_PATH} > /dev/null
+pushd "${WORKING_PATH}" > /dev/null || exit
 
-TENSILE_HOST="https://github.com/${TENSILE_FORK}/Tensile.git"
-ROCBLAS_HOST="https://github.com/${ROCBLAS_FORK}/rocBLAS-internal.git"
+TENSILE_HOST="https://github.com/${FORK}/Tensile.git"
+ROCBLAS_HOST="https://github.com/${FORK}/rocBLAS-internal.git"
```

Comment:
Do we want to expose rocBLAS-internal repo as the default to the public? It seems that this should be hidden from view.

Reply:
Internal was already in place when I started working on these, so I just left it. However, with the new ability to specify an existing local rocBLAS path, I don't see any reason why the ability to explicitly use the internal repo is needed in these scripts. If the internal (or any other version) is needed, the user can just prep rocBLAS ahead of time themselves.